### ISSUE TITLE
Feature to rename movie titles according a found string in the filename ("OmU", "OV", "OF")

### ISSUE THREAD

**Issue by gitttt54** [Casvt/Plex-scripts#136]
Hello,

>In what file should the feature be added?

Dunno.

>What is the feature request?

Short:
Rename (or add a suffix) the movie title in Plex when the filename of that movie has a user-definable string in it.

Long:

We have movies in foreign language only, say, japanese movies with japanese audio, and only german or english or russian subtitles to understand the movie. Plex Users often only see the Movie title and think "Hey, that is a great movie I always wanted to watch", but then are disappointed, because the movie does not contain audio in their native language, but only subtitles. 
This is the reason, why I'd like to have a script which automaticly "tells" the plex user: "Hey, this is a movie in a foreign language, and you have to watch it with subtitles only". This "telling" is done by changing the movie title accordingly ;-D

[Here](https://leipglo.com/2015/08/04/moviesomu-english-language-and-other-foreign-movies-in-the-original-in-leipzig-august-5-12/) is a Text which explains it a little bit more:
_Here’s our regular listing of films being shown in English and other original foreign languages (with labels such as **OmU**, **OF** and **OV**) at Leipzig cinemas for this movie week. We are now running the list each Tuesday. Although not dubbed, the movies often come with German subtitles, so you can still access the movie if you don’t know the original language but know some German._

The MKV files have set the language correctly to, say, japanese language, so Plex users can see what spoken language the movie is before starting it, and can select their preferred subtitle (if not hardcoded).

Unfortunately, the movietitles do not show, that this movie is a foreign movie with only subtitles on the viewers language,
and therefore, I need to rename all the movies from, say `Japanese Movie` to `Japanese Movie (OmU)`, when the Filename contains a string `(OmU)`. 
So, if the filename is `Japanese.Movie.1999.(OmU).mp3.xvid.mkv`, the Script should rename the movietitle from `Japanese Movie` to `Japanese Movie (OmU)` (or add a suffix `(OmU)`, if this will be easier....).

Important: it must be checked, if the movietitle already has an `(OmU)`, 
because we don't want titles like `Japanese Movie (OmU) (OmU) (OmU)`

Here is a list with movies for more examples:

```
Filename: .../Swing Girls (2004)/Swing.Girls.(OmU).(2004).jpn.MP3.360p.DVD.XviD.mkv 
Movie Title before: Swing Girls
Movie Title after: Swing Girls (OmU)

Filename: .../Udon (2006)/Udon.(OmU).(2006).jpn.AC3.570p.DVD.XviD.mkv 
Movie Title before: Udon (OmU)
Movie Title after : Udon (OmU)

Filename: .../Like Father, like Son (2013)/Like.Father,.like.Son.(2013).ger.AC3.570p.DVD.h264.mkv
Movie Title before: Like Father, like Son
Movie Title after : Like Father, like Son

Filename: .../Ayurveda - Art of Being (2001)/Ayurveda.-.Art.of.Being.(English.Subbed).(2001).hindi.MP3.570p.DVD.XVID.mkv 
Movie Title before: Ayurveda - Art of Being
Movie Title after : Ayurveda - Art of Being

Filename: .../Ayurveda - Art of Being (2001)/Ayurveda.-.Art.of.Being.(OmU).(2001).hindi.MP3.570p.DVD.XVID.mkv 
Movie Title before: Ayurveda - Art of Being
Movie Title after : Ayurveda - Art of Being (OmU)
```

In the script I'd like to define 2 main options:

the _first_ string which the script is looking for in the filename
and a _second_ string which is used to rename or to put a suffix on the filename.

So I could call the script like: 
`superscript.py --StringToLookFor "(OmU)" --StringToAppendToMovieTitle "(OmU)"`
or
`superscript.py --StringToLookFor "(English.Subbed)" --StringToAppendToMovieTitle "(English Only)"`

I dont' know, if it is a good idea to lock the movie title field after renaming, but it could be optional in the script by an additional option like `--LockAfterRenaming`.

Also, with this, I don't know, if the script should or should not rename locked titles fields, but I guess, a "force rename locked title field" option would then also be an option to think about.

I personally lock all the fields, I have manually changed, to have Plex not change them again, but with this Script, I really would like to change those title fields, regardless of its locked state. So I don't know, what is the best practice here.

Ok, I hope, you have no trouble understand why I'd like to have this script and how it should work.

I will be glad to answer questions, if there are any.

Thank you!

**Comment by Casvt** [comment]
Super clear. Basically: "if ABC is in the filename, add XYZ to the plex title". Then also give options for handling stuff like locked state. Maybe use a dictionary inside the file to define the "links"? Like this:
```python
links = {
	'(OmU)': '(OmU)',
	'(English.Subbed)': '(English Only)',
	'ABC': 'XYZ'
}
```
Because this way you can define multiple "links"/"connections" at the same time and this way rename all your titles at the same time instead of having to run the script again and again for every "link". And even have the script add multiple strings to the title if there are multiple different matches in the filename. Image this usecase:
```python
links = {
	'(en)': 'English',
	'(nl)': 'Dutch',
	'(it)': 'Italian'
}
```
Imagine a massive dict like this with every language and that way you can just add the language found in the filename to the title in plex. Way easier than running the script like 250+ times for every language.

This is actually really easy haha. You got it \:\)

**Comment by gitttt54** [comment]
Exactly the way I wanted it to be! 

Just for clarification,

if I have a Movie named `Fresh Movie` in Plex with Filename `Fresh.Movie.(1999).(OmU).(Remastered).xvid.mkv`
and define, say, 
```
links = {
	'(OmU)': '(OmU)',
	'(Remastered)': '(Remastered)'
}
```
The resulting Movie Title be `Fresh Movie (OmU) (Remastered)` 
with only one go of the script? 
If yes, that really would be great! Thank you!

P.S.
Maybe its nice to have an option to ignore the case in the search string? 
Just an idea, because I now saw some Filenames with `(omu)` or `(remastered)` and not only `(OmU)` or `(Remastered)`. 
But this could  be solved also with renaming the files from the command line or any renamer program too.

**Comment by Casvt** [comment]
> If yes, that really would be great! Thank you!

Yes

And about the case insensitivity: I'll just add an option to make the triggers (the keys in the dict) case insensitive.
